### ISSUE TITLE
First Load JS not decreasing if i use dyamic import in external package.

### ISSUE THREAD

**Issue by its-mXc** [vercel/next.js#28125]
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

12.18.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

Hi, we have 3 projects using next js and to reduce duplication of code we have created a component library(npm package) with next as a dev dependency to reduce it. If use use dynamic import the component library package file in my next js project it is working as intended but if i use dynamic import in my component library it is not working. Instead everything is fetched in single js.

I suspect dynamic can be used at project level only and disabled for node_modules and it makes sense. Is there a way to force to use dyanmic for specific package in webpack/next.config ?

Thank you

### Expected Behavior

Is there a possibility to make it work ?. Or i would have to import dynamically at project level

### To Reproduce

```
// @npm-component-library/sampleproject.tsx
import React from "react";
import LightComponent from "../LightComponent";
const HeavyComponent = dynamic(() => import("../HeavyComponent"), {
  ssr: false,
});

const SampleProject = () => {
  return (
    <>
      <LightComponent />
      <HeavyComponent />
    </>
  );
};

export default SampleProject
```

```
// nextjs-project/pages/index.tsx
import React from "react";
import SampleProject  from "@npm-component-library/sampleproject"

const Home = () => {
  return <SampleProject />;
};

export default Home;
```

**Comment by VladimirMikulic** [comment]
@its-mXc just passing by here - it really depends whether dynamic import makes sense for your use case.
The example that you have given does more harm to user experience (increased CLS) than good since your components are rendered immediately.

Use case in which dynamic import does make sense would be where some package/component is needed only for a given action.
Let's say you want to display confetti to the user after the product purchase, it would makes sense to load that confetti library dynamically.

**Comment by its-mXc** [comment]
@VladimirMikulic You are right and i understand your concern. In my case it is actually a pdf viewer which is needed for given action. I should have worded the example better

**Comment by VladimirMikulic** [comment]
@its-mXc that makes sense. Have you maybe tried using React's built-in `lazy()` function in that library instead of `dynamic` since Next.js is listed as a dev dependency?

Another potential solution could be to list Next.js in that library as a peer dependency, meaning that the project that uses the library needs to have Next.js installed.

**Comment by its-mXc** [comment]
@VladimirMikulic I forgot to mention next is already a peer dependency.Thanks for the React's lazy suggestion,  the last fix will be use React's lazy if not able to make dynamic working

**Comment by its-mXc** [comment]
Just Wanted to let know, React.lazy also didn't work : (

**Comment by VladimirMikulic** [comment]
That's unfortunate! I guess the only solution would be to dynamically import the library or expose the parts of it to be dynamically imported.

**Comment by its-mXc** [comment]
I got it working.

I figured if it will start working if i move it out from the node_modules to root project and change the alias in webpack config to the new path instead of node_modules one, it will start working. It did not. after investigating more i found it because i was compiling my package code to common js. The dynamic import works if it is es6 import format(weird). So i had to change the type of my package to module and compile it in es6.

But next js does not support es module packages no i had to use next-transpile-modules, which i suspect if coping the node_modules to project and changing the alias. But it is working now

I will close this issue as i am not sure if this is a bug or next js team wanted to use dynamic this way.

Thanks @VladimirMikulic for all the help and suggestions : ). Hope the issue helps someone.
